docs/src/experiments/spherical_delaunay_stereographic.jl

```diff
@@ -0,0 +1,332 @@
+    far_value = 1e6 * sqrt(max_radius)
+
+    if !isempty(really_far_idxs)
+        triangulation_points[really_far_idxs] .= (Point2(far_value, 0.0),)
+    end
+
+    boundary_points = reverse([
+        (-far_value, -far_value),
+        (-far_value, far_value),
+        (far_value, far_value),
+        (far_value, -far_value),
+        (-far_value, -far_value),
+    ])
+
+    # boundary_nodes, pts = DelTri.convert_boundary_points_to_indices(boundary_points; existing_points=triangulation_points)
+
+    # triangulation = DelTri.triangulate(triangulation_points; boundary_nodes)
+    triangulation = DelTri.triangulate(triangulation_points)
+    #  for diagnostics, try `fig, ax, sc = triplot(triangulation, show_constrained_edges=true, show_convex_hull=true)`
+
+    # First, get all the "solid" faces, ie, faces not attached to boundary nodes
+    boundary_faces = Iterators.filter(Base.Fix1(DelTri.is_boundary_triangle, triangulation), DelTri.each_solid_triangle(triangulation)) |> collect
+    faces = map(facetype, DelTri.each_solid_triangle(triangulation))
+
+    for ghost_face in DelTri.each_ghost_triangle(triangulation)
+        push!(faces, facetype(map(ghost_face) do i; DelaunayTriangulation.is_ghost_vertex(i) ? pivot_ind : i end))
+    end
+
+    return faces
+end
+
+# These points are known to be good points, i.e., lon, lat, alt
+points = Point3{Float64}.(JSON3.read(read(Downloads.download("https://gist.githubusercontent.com/Fil/6bc12c535edc3602813a6ef2d1c73891/raw/3ae88bf307e740ddc020303ea95d7d2ecdec0d19/points.json"), String)))
+faces = delaunay_triangulate_spherical(points)
+
+# This is the super-cool scrollable 3D globe (though it's a bit deformed... :D)
+f, a, p = Makie.mesh(map(UnitCartesianFromGeographic(), points), faces; color = last.(points), colormap = Reverse(:RdBu), colorrange = (-20, 40), shading = NoShading)
+
+# We can also replicate the observable notebook almost exactly (just missing ExactPredicates):
+f, a, p = Makie.mesh(points, faces; axis = (; type = GeoAxis, dest = "+proj=bertin1953 +lon_0=-16.5 +lat_0=-42 +x_0=7.93 +y_0=0.09"), color = last.(points), colormap = Reverse(:RdBu), colorrange = (-20, 40), shading = NoShading)
+# Whoops, this doesn't look so good!  Let's try to do this more "manually" instead.
+# We'll use NaturalNeighbours.jl for this, but we first have to reconstruct the triangulation
+# with the same faces, but in a Bertin projection...
+using NaturalNeighbours
+lonlat2bertin = Proj.Transformation(GFT.EPSG(4326), GFT.ProjString("+proj=bertin1953 +type=crs +lon_0=-16.5 +lat_0=-42 +x_0=7.93 +y_0=0.09"); always_xy = true)
+lons = LinRange(-180, 180, 300)
+lats = LinRange(-90, 90, 150)
+bertin_points = lonlat2bertin.(lons, lats')
+
+projected_points = GO.reproject(GO.tuples(points), source_crs = GFT.EPSG(4326), target_crs = "+proj=bertin1953 +lon_0=-16.5 +lat_0=-42 +x_0=7.93 +y_0=0.09")
+
+ch = DelTri.convex_hull(projected_points) # assumes each point is in the triangulation
```

Comment:
@asinghvi17 `projected_points` is three-dimensional. The `convex_hull` function is treating it by ignoring the `z` coordinate of each point. What is supposed to be the input here? This is the issue you're having with NaturalNeighbours - the triangulation is very invalid.

![image](https://github.com/user-attachments/assets/bccd67d5-1d60-4243-b460-efd3a18ed615)

Reply:
`projected_points`' z coordinate only carries metadata, so it's fine for it to be three-dimensional.  That being said I haven't checked the triangulation in stereographic space, it might have issues there too.

Reply:
On main the boundary points generate degenerate triangles, I guess I can remove those?

Reply:
I'm surprised that it's generating triangles where some of the vertices repeat..

Reply:
That's not entirely unexpected, since the triangulation is computed in stereographic space (where the triangulation generated _is_ planar, correct, etc) and not in the projected space that we are currently in.

Reply:
I see. I'm not familiar enough with stereographic projections, I thought they would preserve the structure. I don't see a good way to work with the 2D triangulation in that case, since it isn't actually a triangulation.

Reply:
Yes, that would be a problem.  Maybe it's best to create some kind of spherical triangulation type, which can also handle spherical Voronoi?  That sounds like a lot of effort though.

Reply:
I don't really know how you would get around this issue. It just seems like a bug somewhere to me.. the projection isn't an isometry but it should be conformal, right? I think conformal maps would show up better than what we see in the image above, e.g. it should preserve angles. The crossings seem to come only from boundary vertices (from a visual check anyway, who knows about the interior) which suggests you're missing something when handling only the interior triangles.

Reply:
Ah, to clarify - the triangulation is done in stereographic space, where the triangulation would be completely valid.  I'm trying to interpolate in the Bertin 1953 projection, which doesn't have the same conformal property as the stereographic projection.  I guess what I was trying to do is actually the incorrect thing, and I should be figuring out how to interpolate on the sphere instead...

Reply:
I see.. my thinking was that you can do everything on the valid triangulation and then lift it back up to the sphere. I'll probably have to look more at the algorithm itself to understand.

Reply:
At least for interpolation, I would expect that to have issues when getting the values of points near the chosen pivot point in spherical space - there's always a point which wouldn't be connected to the triangulation in stereographic space, and neighbours in spherical space would be separated by almost the entire domain of points.

The first problem could be solved by injecting a rectangular boundary, but I'm not really sure how to deal with the second problem.